libraries/linux_updates.rb

```diff
@@ -208,10 +208,17 @@ def packages
   end
 
   def updates
-    rhel_updates = <<~PRINT_JSON
-      #!/bin/sh
-      python -c 'import sys; sys.path.insert(0, "/usr/share/yum-cli"); import cli; ybc = cli.YumBaseCli(); ybc.setCacheDir("/tmp"); list = ybc.returnPkgLists(["updates"]);res = ["{\\"name\\":\\""+x.name+"\\", \\"version\\":\\""+x.version+"-"+x.release+"\\",\\"arch\\":\\""+x.arch+"\\",\\"repository\\":\\""+x.repo.id+"\\"}" for x in list.updates]; print "{\\"available\\":["+",".join(res)+"]}"'
-    PRINT_JSON
+    rhel_updates = if os.release.to_i > 7
```

Comment:
Did this work for you? I had to change it like this to get it working:
```suggestion
    rhel_updates = if @inspec.os.release.to_i > 7
```

With your code I got the following error:

```
  ×  verify-patches: Operating system is up-to-date
     ×  Linux Update is expected to be uptodate
     undefined local variable or method `os' for #<#<Class:0x000055c3d77837a0>::RHELUpdateFetcher:0x000055c3d7730848 @inspec=Inspec::Backend::Class @transport=Train::Transports::Docker::Connection>
```

Using:

```
> cinc-auditor -v
5.18.14
```

Reply:
I hadn't had a chance to test it in an inspec run, I had did internal testing to figure out the python code. I just added a vagrant and dokken kitchen file and was able to test successfully using the latest chef-workstation. 
❯ chef -v
Chef Workstation version: 22.7.1006
Cookstyle version: 7.32.1
Chef Infra Client version: 17.10.0
Chef InSpec version: 4.56.20
Chef CLI version: 5.6.1
Chef Habitat version: 1.6.420
Test Kitchen version: 3.3.1

kitchen verify default-centos-8 failed trying to create and load appstream. But almalinux-8 and centos-7 did work.

Reply:
```
❯ kitchen list
/Users/corey.hemminger/Documents/github/personal/chef/.chef/gem/ruby/3.0.0/gems/kitchen-docker-2.13.0/lib/docker/version.rb:20: warning: already initialized constant Docker::VERSION
/opt/chef-workstation/embedded/lib/ruby/gems/3.0.0/gems/docker-api-2.2.0/lib/docker/version.rb:3: warning: previous definition of VERSION was here
Instance             Driver  Provisioner  Verifier  Transport  Last Action    Last Error
default-ubuntu-1604  Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-ubuntu-1804  Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-ubuntu-2004  Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-ubuntu-2204  Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-centos-6     Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-centos-7     Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-centos-8     Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-almalinux-8  Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-almalinux-9  Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-debian-8     Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-debian-9     Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-debian-10    Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
default-debian-11    Docker  ChefInfra    Inspec    Docker     <Not Created>  <None>
❯ kitchen verify default-centos-7
-----> Starting Test Kitchen (v3.3.1)
/Users/corey.hemminger/Documents/github/personal/chef/.chef/gem/ruby/3.0.0/gems/kitchen-docker-2.13.0/lib/docker/version.rb:20: warning: already initialized constant Docker::VERSION
/opt/chef-workstation/embedded/lib/ruby/gems/3.0.0/gems/docker-api-2.2.0/lib/docker/version.rb:3: warning: previous definition of VERSION was here
-----> Creating <default-centos-7>...
       #1 [internal] load build definition from Dockerfile-kitchen20220913-55025-8gp7m
       #1 sha256:aa1ca390b087a3053f8bdfe72f2258b4034cdd32dac7f91c786192b3a6746cd4
       #1 transferring dockerfile: 1.55kB done
       #1 DONE 0.0s
       
       #2 [internal] load .dockerignore
       #2 sha256:db3e03fc5b97e78df7c97b851438e186b4e152ee5a1d5b41922302af8f8e5ef1
       #2 transferring context: 2B done
       #2 DONE 0.0s
       
       #3 [internal] load metadata for docker.io/library/centos:7
       #3 sha256:30875b35a89c8e8a29cd7cf120689bb68cdab8d769419707e07138dfe977d237
       #3 ...
       
       #4 [auth] library/centos:pull token for registry-1.docker.io
       #4 sha256:e4253f5016e14d5ddc4ecdd43572f29c7591507d02e5bf253c1af7f69f436cc4
       #4 DONE 0.0s
       
       #3 [internal] load metadata for docker.io/library/centos:7
       #3 sha256:30875b35a89c8e8a29cd7cf120689bb68cdab8d769419707e07138dfe977d237
       #3 DONE 1.1s
       
       #5 [ 1/16] FROM docker.io/library/centos:7@sha256:c73f515d06b0fa07bb18d8202035e739a494ce760aa73129f60f4bf2bd22b407
       #5 sha256:8c82933c032a7bd124a27346cd3e10fe831376502aac9e75ecc612ae5f699533
       #5 resolve docker.io/library/centos:7@sha256:c73f515d06b0fa07bb18d8202035e739a494ce760aa73129f60f4bf2bd22b407 done
       #5 sha256:c73f515d06b0fa07bb18d8202035e739a494ce760aa73129f60f4bf2bd22b407 1.20kB / 1.20kB done
       #5 sha256:dead07b4d8ed7e29e98de0f4504d87e8880d4347859d839686a31da35a3b532f 529B / 529B done
       #5 sha256:eeb6ee3f44bd0b5103bb561b4c16bcb82328cfe5809ab675bb17ab3a16c517c9 2.75kB / 2.75kB done
       #5 DONE 0.0s
       
       #6 [ 2/16] RUN yum clean all
       #6 sha256:71cf723a3c56bde7e9090a2e3e97224ac871227303e5cb556a3e22b5ea0cf25f
       #6 0.287 Loaded plugins: fastestmirror, ovl
       #6 0.369 Cleaning repos: base extras updates
       #6 DONE 0.4s
       
       #7 [ 3/16] RUN yum install -y sudo openssh-server openssh-clients which curl
       #7 sha256:4c309efc4a866edbcd6b4e744539d56a082d00ee549e642c322ab79bca389749
       #7 0.460 Loaded plugins: fastestmirror, ovl
       #7 0.558 Determining fastest mirrors
       #7 0.914  * base: mirrors.usinternet.com
       #7 0.915  * extras: mirror.cs.uwp.edu
       #7 0.915  * updates: atl.mirrors.clouvider.net
       #7 5.577 Resolving Dependencies
       #7 5.578 --> Running transaction check
       #7 5.578 ---> Package curl.x86_64 0:7.29.0-59.el7 will be updated
       #7 5.581 ---> Package curl.x86_64 0:7.29.0-59.el7_9.1 will be an update
       #7 5.590 --> Processing Dependency: libcurl = 7.29.0-59.el7_9.1 for package: curl-7.29.0-59.el7_9.1.x86_64
       #7 5.594 ---> Package openssh-clients.x86_64 0:7.4p1-22.el7_9 will be installed
       #7 5.599 --> Processing Dependency: openssh = 7.4p1-22.el7_9 for package: openssh-clients-7.4p1-22.el7_9.x86_64
       #7 5.601 --> Processing Dependency: fipscheck-lib(x86-64) >= 1.3.0 for package: openssh-clients-7.4p1-22.el7_9.x86_64
       #7 5.602 --> Processing Dependency: libfipscheck.so.1()(64bit) for package: openssh-clients-7.4p1-22.el7_9.x86_64
       #7 5.602 --> Processing Dependency: libedit.so.0()(64bit) for package: openssh-clients-7.4p1-22.el7_9.x86_64
       #7 5.604 ---> Package openssh-server.x86_64 0:7.4p1-22.el7_9 will be installed
       #7 5.609 --> Processing Dependency: libwrap.so.0()(64bit) for package: openssh-server-7.4p1-22.el7_9.x86_64
       #7 5.610 ---> Package sudo.x86_64 0:1.8.23-10.el7_9.2 will be installed
       #7 5.613 ---> Package which.x86_64 0:2.20-7.el7 will be installed
       #7 5.614 --> Running transaction check
       #7 5.614 ---> Package fipscheck-lib.x86_64 0:1.4.1-6.el7 will be installed
       #7 5.615 --> Processing Dependency: /usr/bin/fipscheck for package: fipscheck-lib-1.4.1-6.el7.x86_64
       #7 5.616 ---> Package libcurl.x86_64 0:7.29.0-59.el7 will be updated
       #7 5.617 ---> Package libcurl.x86_64 0:7.29.0-59.el7_9.1 will be an update
       #7 5.620 ---> Package libedit.x86_64 0:3.0-12.20121213cvs.el7 will be installed
       #7 5.621 ---> Package openssh.x86_64 0:7.4p1-22.el7_9 will be installed
       #7 5.623 ---> Package tcp_wrappers-libs.x86_64 0:7.6-77.el7 will be installed
       #7 5.624 --> Running transaction check
       #7 5.624 ---> Package fipscheck.x86_64 0:1.4.1-6.el7 will be installed
       #7 5.689 --> Finished Dependency Resolution
       #7 5.693 
       #7 5.693 Dependencies Resolved
       #7 5.695 
       #7 5.695 ================================================================================
       #7 5.695  Package               Arch       Version                     Repository   Size
       #7 5.695 ================================================================================
       #7 5.695 Installing:
       #7 5.695  openssh-clients       x86_64     7.4p1-22.el7_9              updates     655 k
       #7 5.695  openssh-server        x86_64     7.4p1-22.el7_9              updates     459 k
       #7 5.695  sudo                  x86_64     1.8.23-10.el7_9.2           updates     843 k
       #7 5.695  which                 x86_64     2.20-7.el7                  base         41 k
       #7 5.695 Updating:
       #7 5.695  curl                  x86_64     7.29.0-59.el7_9.1           updates     271 k
       #7 5.695 Installing for dependencies:
       #7 5.695  fipscheck             x86_64     1.4.1-6.el7                 base         21 k
       #7 5.695  fipscheck-lib         x86_64     1.4.1-6.el7                 base         11 k
       #7 5.695  libedit               x86_64     3.0-12.20121213cvs.el7      base         92 k
       #7 5.695  openssh               x86_64     7.4p1-22.el7_9              updates     510 k
       #7 5.695  tcp_wrappers-libs     x86_64     7.6-77.el7                  base         66 k
       #7 5.695 Updating for dependencies:
       #7 5.695  libcurl               x86_64     7.29.0-59.el7_9.1           updates     223 k
       #7 5.695 
       #7 5.695 Transaction Summary
       #7 5.695 ================================================================================
       #7 5.695 Install  4 Packages (+5 Dependent packages)
       #7 5.695 Upgrade  1 Package  (+1 Dependent package)
       #7 5.695 
       #7 5.695 Total download size: 3.1 M
       #7 5.696 Downloading packages:
       #7 5.697 Delta RPMs disabled because /usr/bin/applydeltarpm not installed.
       #7 5.809 Public key for fipscheck-lib-1.4.1-6.el7.x86_64.rpm is not installed
       #7 5.809 warning: /var/cache/yum/x86_64/7/base/packages/fipscheck-lib-1.4.1-6.el7.x86_64.rpm: Header V3 RSA/SHA256 Signature, key ID f4a80eb5: NOKEY
       #7 5.919 Public key for curl-7.29.0-59.el7_9.1.x86_64.rpm is not installed
       #7 6.180 --------------------------------------------------------------------------------
       #7 6.180 Total                                              6.5 MB/s | 3.1 MB  00:00     
       #7 6.182 Retrieving key from file:///etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
       #7 6.187 Importing GPG key 0xF4A80EB5:
       #7 6.187  Userid     : "CentOS-7 Key (CentOS 7 Official Signing Key) <security@centos.org>"
       #7 6.187  Fingerprint: 6341 ab27 53d7 8a78 a7c2 7bb1 24c6 a8a7 f4a8 0eb5
       #7 6.187  Package    : centos-release-7-9.2009.0.el7.centos.x86_64 (@CentOS)
       #7 6.187  From       : /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7
       #7 6.202 Running transaction check
       #7 6.224 Running transaction test
       #7 6.238 Transaction test succeeded
       #7 6.239 Running transaction
       #7 6.265   Installing : fipscheck-1.4.1-6.el7.x86_64                                1/13 
       #7 6.277   Installing : fipscheck-lib-1.4.1-6.el7.x86_64                            2/13 
       #7 6.801   Installing : openssh-7.4p1-22.el7_9.x86_64                               3/13 
       #7 6.832   Updating   : libcurl-7.29.0-59.el7_9.1.x86_64                            4/13 
       #7 7.073   Installing : tcp_wrappers-libs-7.6-77.el7.x86_64                         5/13 
       #7 7.310   Installing : libedit-3.0-12.20121213cvs.el7.x86_64                       6/13 
       #7 7.586   Installing : openssh-clients-7.4p1-22.el7_9.x86_64                       7/13 
       #7 7.857   Installing : openssh-server-7.4p1-22.el7_9.x86_64                        8/13 
       #7 8.091   Updating   : curl-7.29.0-59.el7_9.1.x86_64                               9/13 
       #7 8.171   Installing : sudo-1.8.23-10.el7_9.2.x86_64                              10/13 
       #7 8.390   Installing : which-2.20-7.el7.x86_64                                    11/13 
       #7 8.596 install-info: No such file or directory for /usr/share/info/which.info.gz
       #7 8.605   Cleanup    : curl-7.29.0-59.el7.x86_64                                  12/13 
       #7 8.614   Cleanup    : libcurl-7.29.0-59.el7.x86_64                               13/13 
       #7 8.843   Verifying  : curl-7.29.0-59.el7_9.1.x86_64                               1/13 
       #7 8.855   Verifying  : fipscheck-lib-1.4.1-6.el7.x86_64                            2/13 
       #7 8.861   Verifying  : openssh-server-7.4p1-22.el7_9.x86_64                        3/13 
       #7 8.869   Verifying  : fipscheck-1.4.1-6.el7.x86_64                                4/13 
       #7 8.876   Verifying  : which-2.20-7.el7.x86_64                                     5/13 
       #7 8.882   Verifying  : libedit-3.0-12.20121213cvs.el7.x86_64                       6/13 
       #7 8.888   Verifying  : openssh-clients-7.4p1-22.el7_9.x86_64                       7/13 
       #7 8.896   Verifying  : sudo-1.8.23-10.el7_9.2.x86_64                               8/13 
       #7 8.902   Verifying  : tcp_wrappers-libs-7.6-77.el7.x86_64                         9/13 
       #7 8.908   Verifying  : libcurl-7.29.0-59.el7_9.1.x86_64                           10/13 
       #7 8.915   Verifying  : openssh-7.4p1-22.el7_9.x86_64                              11/13 
       #7 8.921   Verifying  : libcurl-7.29.0-59.el7.x86_64                               12/13 
       #7 8.922   Verifying  : curl-7.29.0-59.el7.x86_64                                  13/13 
       #7 8.943 
       #7 8.943 Installed:
       #7 8.943   openssh-clients.x86_64 0:7.4p1-22.el7_9                                       
       #7 8.943   openssh-server.x86_64 0:7.4p1-22.el7_9                                        
       #7 8.943   sudo.x86_64 0:1.8.23-10.el7_9.2                                               
       #7 8.943   which.x86_64 0:2.20-7.el7                                                     
       #7 8.943 
       #7 8.943 Dependency Installed:
       #7 8.943   fipscheck.x86_64 0:1.4.1-6.el7            fipscheck-lib.x86_64 0:1.4.1-6.el7  
       #7 8.943   libedit.x86_64 0:3.0-12.20121213cvs.el7   openssh.x86_64 0:7.4p1-22.el7_9     
       #7 8.943   tcp_wrappers-libs.x86_64 0:7.6-77.el7    
       #7 8.943 
       #7 8.943 Updated:
       #7 8.943   curl.x86_64 0:7.29.0-59.el7_9.1                                               
       #7 8.943 
       #7 8.943 Dependency Updated:
       #7 8.943   libcurl.x86_64 0:7.29.0-59.el7_9.1                                            
       #7 8.943 
       #7 8.943 Complete!
       #7 DONE 9.1s
       
       #8 [ 4/16] RUN [ -f "/etc/ssh/ssh_host_rsa_key" ] || ssh-keygen -t rsa -f /etc/ssh/ssh_host_rsa_key -N ''
       #8 sha256:34434e9923bc3493a00d8eb75857a4e5eeed3738171b96020dc8b6a57e80f372
       #8 0.281 Generating public/private rsa key pair.
       #8 0.281 Your identification has been saved in /etc/ssh/ssh_host_rsa_key.
       #8 0.281 Your public key has been saved in /etc/ssh/ssh_host_rsa_key.pub.
       #8 0.281 The key fingerprint is:
       #8 0.281 SHA256:Qvq2sgYnwcRGkHYUD/QzbT44yQGtzlWUQ1qAnoBnVzU root@buildkitsandbox
       #8 0.281 The key's randomart image is:
       #8 0.281 +---[RSA 2048]----+
       #8 0.281 |o=+*+o==E        |
       #8 0.281 |oo*o*.++ .       |
       #8 0.281 |.*+.oBoo.        |
       #8 0.281 |  o+.+O          |
       #8 0.281 |  o.o=.oS        |
       #8 0.281 |  oo.....        |
       #8 0.281 |   +  o          |
       #8 0.281 |    o. .         |
       #8 0.281 |   ..o.          |
       #8 0.281 +----[SHA256]-----+
       #8 DONE 0.3s
       
       #9 [ 5/16] RUN [ -f "/etc/ssh/ssh_host_dsa_key" ] || ssh-keygen -t dsa -f /etc/ssh/ssh_host_dsa_key -N ''
       #9 sha256:d3087949235deeb57cbef60327930ad9112efbacf78e2eb049710739985b5cd5
       #9 0.390 Generating public/private dsa key pair.
       #9 0.390 Your identification has been saved in /etc/ssh/ssh_host_dsa_key.
       #9 0.390 Your public key has been saved in /etc/ssh/ssh_host_dsa_key.pub.
       #9 0.390 The key fingerprint is:
       #9 0.390 SHA256:9nB02+LUBwjIbdFc3RIeGmcGryuz44ZJ2zSIk2h1nmw root@buildkitsandbox
       #9 0.390 The key's randomart image is:
       #9 0.390 +---[DSA 1024]----+
       #9 0.390 |       . oo+ooBo.|
       #9 0.390 |        o o.oX..o|
       #9 0.390 |         .. + +. |
       #9 0.390 |      . .. . = . |
       #9 0.390 |     o *So. = o .|
       #9 0.390 |    o +.E+oo o . |
       #9 0.390 |   .   + *+.o    |
       #9 0.390 |        + ++     |
       #9 0.390 |         oo.     |
       #9 0.390 +----[SHA256]-----+
       #9 DONE 0.4s
       
       #10 [ 6/16] RUN if ! getent passwd kitchen; then                   useradd -d /home/kitchen -m -s /bin/bash -p '*' kitchen;                 fi
       #10 sha256:7d40846e1f059d474da14f97fb779fa73b9fd66a574dbf57ebf1fea53cb5666b
       #10 DONE 0.3s
       
       #11 [ 7/16] RUN echo "kitchen ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers.d/kitchen
       #11 sha256:146a001291b8494c245e46baa9033172317e3306399c08509ddf798d5acdb35d
       #11 DONE 0.4s
       
       #12 [ 8/16] RUN echo "Defaults !requiretty" >> /etc/sudoers.d/kitchen
       #12 sha256:4fc9d3fbdb3eadb60bacad40acc0261ed4b6a1a4f4c1ecf1bd169186ffe72179
       #12 DONE 0.2s
       
       #13 [ 9/16] RUN mkdir -p /home/kitchen/.ssh
       #13 sha256:347573cc247e48ac705a03bb72c7614ce84a03797fed5de999a90a0f96e1a17e
       #13 DONE 0.4s
       
       #14 [10/16] RUN chown -R kitchen /home/kitchen/.ssh
       #14 sha256:1edd495b3ab1c582df76ed950bb7ed9424f82ab80ca95b9b3068353bb7134519
       #14 DONE 0.3s
       
       #15 [11/16] RUN chmod 0700 /home/kitchen/.ssh
       #15 sha256:d35e4a151f7b6441f870314edf061e0bd34d54c2342174fa6a424210bab35331
       #15 DONE 0.3s
       
       #16 [12/16] RUN touch /home/kitchen/.ssh/authorized_keys
       #16 sha256:81b1d6952a8d51c90eededaf64c78d2fc044230f4ebce60f77cb7ced901a4901
       #16 DONE 0.4s
       
       #17 [13/16] RUN chown kitchen /home/kitchen/.ssh/authorized_keys
       #17 sha256:52421c6005ebdcb9327394bec58fa6ea357380db163d562c5d79a0d571c0c3c9
       #17 DONE 0.4s
       
       #18 [14/16] RUN chmod 0600 /home/kitchen/.ssh/authorized_keys
       #18 sha256:5c7354eca44ab3e9212aeb16baff0daffdac0997869d5debc6ee0ae31fbbc82a
       #18 DONE 0.3s
       
       #19 [15/16] RUN mkdir -p /run/sshd
       #19 sha256:861021123aa8b15caca54031bf7651fcb23d2b3e0321bc5686bb6c17b2be475a
       #19 DONE 0.4s
       
       #20 [16/16] RUN echo ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCeNNln+Jjid1GYvWtUoCS6eg+VQn2ODjaH8YnDElUsFmVFTNd6Z8tWRyCnnuA16GXm2CKCsaxv+laisXldVfO0Fz0jW6DjW0kxzB6dejRfYh8RmGkTUbXE1BR5fxl/cx9lVMM9LMMco94QLjlZzmY+LzUodoI7IK5w4Dj832E0tDn8jXB6Jm0Fezahhq2xCHcyYotI8muof0S/MhiRf9xwrR+k8Ruhw/H3goYK2v5qjHJoCAx/kW+V9xU5nK/L+uTV4f4JyQ98uPvUwWbZQdFugyfdDzJCf3f6/11F3Wbay09unrEBamzgGJnAtoHyNytc26br0lMXwi/QMOIMbNxT kitchen_docker_key >> /home/kitchen/.ssh/authorized_keys
       #20 sha256:b7331fc12ecde2469d4948df2948f83ff31037e28408069ec48289b9b2455a78
       #20 DONE 0.4s
       
       #21 exporting to image
       #21 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
       #21 exporting layers
       #21 exporting layers 1.6s done
       #21 writing image sha256:3b8dea785885ae55cee753d512eb498c211bfcd11c974afd4f6e693cc1e0ba99 done
       #21 DONE 1.6s
       2c8b65eee38c60d428f2dab8164417894804fa34dc827b0246a0f9dae11ab068
       0.0.0.0:50130
       Finished creating <default-centos-7> (0m17.75s).
-----> Converging <default-centos-7>...
       Preparing files for transfer
       Preparing dna.json
       Policyfile, Berksfile, cookbooks/, or metadata.rb not found so Chef Infra Client will run, but do nothing. Is this intended?
       Removing non-cookbook files before transfer
       Preparing validation.pem
       Preparing client.rb
       [Docker] Executing command on container
-----> Installing Chef install only if missing package
       Downloading https://omnitruck.chef.io/install.sh to file /tmp/install.sh
       Trying curl...
       Download complete.
       el 7 x86_64
       Getting information for chef stable  for el...
       downloading https://omnitruck.chef.io/stable/chef/metadata?v=&p=el&pv=7&m=x86_64
         to file /tmp/install.sh.29/metadata.txt
       trying curl...
       sha1	8a598cce806d6a1b7f1c911e2f1f111498fd748a
       sha256	037bfdcab5000a1fbaa9c11dda8a22bf536230cc84398e4c606e77b8136e63ad
       url	https://packages.chef.io/files/stable/chef/17.10.3/el/7/chef-17.10.3-1.el7.x86_64.rpm
       version	17.10.3
       downloaded metadata file looks valid...
       downloading https://packages.chef.io/files/stable/chef/17.10.3/el/7/chef-17.10.3-1.el7.x86_64.rpm
         to file /tmp/install.sh.29/chef-17.10.3-1.el7.x86_64.rpm
       trying curl...
       Comparing checksum with sha256sum...
       
       WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
       
       You are installing a package without a version pin.  If you are installing
       on production servers via an automated process this is DANGEROUS and you will
       be upgraded without warning on new releases, even to new major releases.
       Letting the version float is only appropriate in test, development or
       CI/CD environments.
       
       WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
       
       Installing chef 
       installing with rpm...
       warning: /tmp/install.sh.29/chef-17.10.3-1.el7.x86_64.rpm: Header V4 DSA/SHA1 Signature, key ID 83ef826a: NOKEY
       Preparing...                          ########################################
       Updating / installing...
       chef-17.10.3-1.el7                    ########################################
       Thank you for installing Chef Infra Client! For help getting started visit https://learn.chef.io
       [Docker] Executing command on container
       Transferring files to <default-centos-7>
       [Docker] Executing command on container
       +---------------------------------------------+
       ✔ 2 product licenses accepted.
       +---------------------------------------------+
       Chef Infra Client, version 17.10.3
       Patents: https://www.chef.io/patents
       Infra Phase starting
       [2022-09-13T17:47:15+00:00] WARN: Plugin Network: unable to detect ipaddress
       [2022-09-13T17:47:15+00:00] ERROR: shard_seed: Failed to get dmi property serial_number: is dmidecode installed?
       Creating a new client identity for default-centos-7 using the validator key.
       Resolving cookbooks for run list: []
       Synchronizing cookbooks:
       Installing cookbook gem dependencies:
       Compiling cookbooks...
       Loading Chef InSpec profile files:
       Loading Chef InSpec input files:
       Loading Chef InSpec waiver files:
       [2022-09-13T17:47:16+00:00] WARN: Node default-centos-7 has an empty run list.
       Converging 0 resources
       
       Running handlers:
       Running handlers complete
       Infra Phase complete, 0/0 resources updated in 01 seconds
       Downloading files from <default-centos-7>
       Finished converging <default-centos-7> (0m10.88s).
-----> Setting up <default-centos-7>...
       Finished setting up <default-centos-7> (0m0.00s).
-----> Verifying <default-centos-7>...
WARNING: Nokogiri was built against libxml version 2.9.4, but has dynamically loaded 2.9.13
       Loaded linux-patch-baseline 
[2022-09-13T12:47:29-05:00] WARN: Cannot find a UUID for your node.

Profile: DevSec Linux Patch Benchmark (linux-patch-baseline)
Version: 0.6.8
Target:  docker://2c8b65eee38c60d428f2dab8164417894804fa34dc827b0246a0f9dae11ab068

  ×  verify-patches: Operating system is up-to-date
     ×  Linux Update is expected to be uptodate
     expected `Linux Update.uptodate?` to be truthy, got false
  ×  patches: All operating system package updates are installed (45 failed)
     ×  System Package expat version is expected to eq "2.1.0-14.el7_9"
     
     expected: "2.1.0-14.el7_9"
          got: "2.1.0-12.el7"
     
     (compared using ==)

     ×  System Package libuuid version is expected to eq "2.23.2-65.el7_9.1"
     
     expected: "2.23.2-65.el7_9.1"
          got: "2.23.2-65.el7"
     
     (compared using ==)

     ×  System Package rpm version is expected to eq "4.11.3-48.el7_9"
     
     expected: "4.11.3-48.el7_9"
          got: "4.11.3-45.el7"
     
     (compared using ==)

     ×  System Package rpm-build-libs version is expected to eq "4.11.3-48.el7_9"
     
     expected: "4.11.3-48.el7_9"
          got: "4.11.3-45.el7"
     
     (compared using ==)

     ×  System Package glibc version is expected to eq "2.17-326.el7_9"
     
     expected: "2.17-326.el7_9"
          got: "2.17-317.el7"
     
     (compared using ==)

     ×  System Package nss-tools version is expected to eq "3.67.0-4.el7_9"
     
     expected: "3.67.0-4.el7_9"
          got: "3.53.1-3.el7_9"
     
     (compared using ==)

     ×  System Package gzip version is expected to eq "1.5-11.el7_9"
     
     expected: "1.5-11.el7_9"
          got: "1.5-10.el7"
     
     (compared using ==)

     ×  System Package glibc-common version is expected to eq "2.17-326.el7_9"
     
     expected: "2.17-326.el7_9"
          got: "2.17-317.el7"
     
     (compared using ==)

     ×  System Package binutils version is expected to eq "2.27-44.base.el7_9.1"
     
     expected: "2.27-44.base.el7_9.1"
          got: "2.27-44.base.el7"
     
     (compared using ==)

     ×  System Package coreutils version is expected to eq "8.22-24.el7_9.2"
     
     expected: "8.22-24.el7_9.2"
          got: "8.22-24.el7"
     
     (compared using ==)

     ×  System Package nss version is expected to eq "3.67.0-4.el7_9"
     
     expected: "3.67.0-4.el7_9"
          got: "3.53.1-3.el7_9"
     
     (compared using ==)

     ×  System Package krb5-libs version is expected to eq "1.15.1-54.el7_9"
     
     expected: "1.15.1-54.el7_9"
          got: "1.15.1-50.el7"
     
     (compared using ==)

     ×  System Package ca-certificates version is expected to eq "2021.2.50-72.el7_9"
     
     expected: "2021.2.50-72.el7_9"
          got: "2020.2.41-70.0.el7_8"
     
     (compared using ==)

     ×  System Package zlib version is expected to eq "1.2.7-20.el7_9"
     
     expected: "1.2.7-20.el7_9"
          got: "1.2.7-18.el7"
     
     (compared using ==)

     ×  System Package nss-softokn version is expected to eq "3.67.0-3.el7_9"
     
     expected: "3.67.0-3.el7_9"
          got: "3.53.1-6.el7_9"
     
     (compared using ==)

     ×  System Package nss-util version is expected to eq "3.67.0-1.el7_9"
     
     expected: "3.67.0-1.el7_9"
          got: "3.53.1-1.el7_9"
     
     (compared using ==)

     ×  System Package glib2 version is expected to eq "2.56.1-9.el7_9"
     
     expected: "2.56.1-9.el7_9"
          got: "2.56.1-7.el7"
     
     (compared using ==)

     ×  System Package vim-minimal version is expected to eq "7.4.629-8.el7_9"
     
     expected: "7.4.629-8.el7_9"
          got: "7.4.629-7.el7"
     
     (compared using ==)

     ×  System Package openssl-libs version is expected to eq "1.0.2k-25.el7_9"
     
     expected: "1.0.2k-25.el7_9"
          got: "1.0.2k-19.el7"
     
     (compared using ==)

     ×  System Package libxml2 version is expected to eq "2.9.1-6.el7_9.6"
     
     expected: "2.9.1-6.el7_9.6"
          got: "2.9.1-6.el7.5"
     
     (compared using ==)

     ×  System Package kpartx version is expected to eq "0.4.9-135.el7_9"
     
     expected: "0.4.9-135.el7_9"
          got: "0.4.9-133.el7"
     
     (compared using ==)

     ×  System Package bind-license version is expected to eq "9.11.4-26.P2.el7_9.9"
     
     expected: "9.11.4-26.P2.el7_9.9"
          got: "9.11.4-26.P2.el7"
     
     (compared using ==)

     ×  System Package nspr version is expected to eq "4.32.0-1.el7_9"
     
     expected: "4.32.0-1.el7_9"
          got: "4.25.0-2.el7_9"
     
     (compared using ==)

     ×  System Package centos-release version is expected to eq "7-9.2009.1.el7.centos"
     
     expected: "7-9.2009.1.el7.centos"
          got: "7-9.2009.0.el7.centos"
     
     (compared using ==)

     ×  System Package nss-sysinit version is expected to eq "3.67.0-4.el7_9"
     
     expected: "3.67.0-4.el7_9"
          got: "3.53.1-3.el7_9"
     
     (compared using ==)

     ×  System Package systemd-libs version is expected to eq "219-78.el7_9.7"
     
     expected: "219-78.el7_9.7"
          got: "219-78.el7"
     
     (compared using ==)

     ×  System Package systemd version is expected to eq "219-78.el7_9.7"
     
     expected: "219-78.el7_9.7"
          got: "219-78.el7"
     
     (compared using ==)

     ×  System Package device-mapper version is expected to eq "1.02.170-6.el7_9.5"
     
     expected: "1.02.170-6.el7_9.5"
          got: "1.02.170-6.el7"
     
     (compared using ==)

     ×  System Package openldap version is expected to eq "2.4.44-25.el7_9"
     
     expected: "2.4.44-25.el7_9"
          got: "2.4.44-22.el7"
     
     (compared using ==)

     ×  System Package device-mapper-libs version is expected to eq "1.02.170-6.el7_9.5"
     
     expected: "1.02.170-6.el7_9.5"
          got: "1.02.170-6.el7"
     
     (compared using ==)

     ×  System Package libblkid version is expected to eq "2.23.2-65.el7_9.1"
     
     expected: "2.23.2-65.el7_9.1"
          got: "2.23.2-65.el7"
     
     (compared using ==)

     ×  System Package python-libs version is expected to eq "2.7.5-92.el7_9"
     
     expected: "2.7.5-92.el7_9"
          got: "2.7.5-89.el7"
     
     (compared using ==)

     ×  System Package xz version is expected to eq "5.2.2-2.el7_9"
     
     expected: "5.2.2-2.el7_9"
          got: "5.2.2-1.el7"
     
     (compared using ==)

     ×  System Package rpm-libs version is expected to eq "4.11.3-48.el7_9"
     
     expected: "4.11.3-48.el7_9"
          got: "4.11.3-45.el7"
     
     (compared using ==)

     ×  System Package util-linux version is expected to eq "2.23.2-65.el7_9.1"
     
     expected: "2.23.2-65.el7_9.1"
          got: "2.23.2-65.el7"
     
     (compared using ==)

     ×  System Package xz-libs version is expected to eq "5.2.2-2.el7_9"
     
     expected: "5.2.2-2.el7_9"
          got: "5.2.2-1.el7"
     
     (compared using ==)

     ×  System Package python version is expected to eq "2.7.5-92.el7_9"
     
     expected: "2.7.5-92.el7_9"
          got: "2.7.5-89.el7"
     
     (compared using ==)

     ×  System Package rpm-python version is expected to eq "4.11.3-48.el7_9"
     
     expected: "4.11.3-48.el7_9"
          got: "4.11.3-45.el7"
     
     (compared using ==)

     ×  System Package libxml2-python version is expected to eq "2.9.1-6.el7_9.6"
     
     expected: "2.9.1-6.el7_9.6"
          got: "2.9.1-6.el7.5"
     
     (compared using ==)

     ×  System Package tzdata version is expected to eq "2022c-1.el7"
     
     expected: "2022c-1.el7"
          got: "2020d-2.el7"
     
     (compared using ==)

     ×  System Package bash version is expected to eq "4.2.46-35.el7_9"
     
     expected: "4.2.46-35.el7_9"
          got: "4.2.46-34.el7"
     
     (compared using ==)

     ×  System Package libsmartcols version is expected to eq "2.23.2-65.el7_9.1"
     
     expected: "2.23.2-65.el7_9.1"
          got: "2.23.2-65.el7"
     
     (compared using ==)

     ×  System Package cyrus-sasl-lib version is expected to eq "2.1.26-24.el7_9"
     
     expected: "2.1.26-24.el7_9"
          got: "2.1.26-23.el7"
     
     (compared using ==)

     ×  System Package nss-softokn-freebl version is expected to eq "3.67.0-3.el7_9"
     
     expected: "3.67.0-3.el7_9"
          got: "3.53.1-6.el7_9"
     
     (compared using ==)

     ×  System Package libmount version is expected to eq "2.23.2-65.el7_9.1"
     
     expected: "2.23.2-65.el7_9.1"
          got: "2.23.2-65.el7"
     
     (compared using ==)



Profile Summary: 0 successful controls, 2 control failures, 0 controls skipped
Test Summary: 0 successful, 46 failures, 0 skipped

❯ kitchen verify default-almalinux-8
-----> Starting Test Kitchen (v3.3.1)
/Users/corey.hemminger/Documents/github/personal/chef/.chef/gem/ruby/3.0.0/gems/kitchen-docker-2.13.0/lib/docker/version.rb:20: warning: already initialized constant Docker::VERSION
/opt/chef-workstation/embedded/lib/ruby/gems/3.0.0/gems/docker-api-2.2.0/lib/docker/version.rb:3: warning: previous definition of VERSION was here
-----> Creating <default-almalinux-8>...
       #1 [internal] load build definition from Dockerfile-kitchen20220913-55243-5m6m1k
       #1 sha256:3e77d7fe36581487a96c14bfd4169c088e35ee1dd33020f85d688f508872e3f9
       #1 transferring dockerfile: 1.55kB done
       #1 DONE 0.0s
       
       #2 [internal] load .dockerignore
       #2 sha256:3658b86d90e7ef4315443ab106288d3708225fd1622f0f121fb88e78b3142b91
       #2 transferring context: 2B done
       #2 DONE 0.0s
       
       #3 [internal] load metadata for docker.io/library/almalinux:8
       #3 sha256:9438fcfbd83b30806a7d2edc65d8aba025310c68d472c423e4e829a424fb5761
       #3 ...
       
       #4 [auth] library/almalinux:pull token for registry-1.docker.io
       #4 sha256:17195c2d69a8b09b686f599ee17c63710943b6c470498b5cd5c6ac39ba19e454
       #4 DONE 0.0s
       
       #3 [internal] load metadata for docker.io/library/almalinux:8
       #3 sha256:9438fcfbd83b30806a7d2edc65d8aba025310c68d472c423e4e829a424fb5761
       #3 DONE 1.1s
       
       #5 [ 1/16] FROM docker.io/library/almalinux:8@sha256:c8369022fa04c7d8f3c0504d4732fabf402ed756753d47291af5798c7910f7bd
       #5 sha256:53ac53f8336cffd32d14058407d0d53056178da91dbbf74aee32a342533d5753
       #5 resolve docker.io/library/almalinux:8@sha256:c8369022fa04c7d8f3c0504d4732fabf402ed756753d47291af5798c7910f7bd done
       #5 sha256:c8369022fa04c7d8f3c0504d4732fabf402ed756753d47291af5798c7910f7bd 975B / 975B done
       #5 sha256:e4d6def1c1e9569f603b3de7d82809fb5b847858872513bd3a11d3b1c10ecf6e 529B / 529B done
       #5 sha256:d1c25ed4de19e4c16adee525e5d547ab9e39c4bbc5296b983b713b5fa8aa76a0 1.48kB / 1.48kB done
       #5 DONE 0.0s
       
       #6 [ 2/16] RUN yum clean all
       #6 sha256:8bc2041703d705b919272a6b121b39deeddbaf6d3983a07f4a0f3294604c236c
       #6 0.346 0 files removed
       #6 DONE 0.4s
       
       #7 [ 3/16] RUN yum install -y sudo openssh-server openssh-clients which curl
       #7 sha256:fed9292c30b1a325a13f3d6618d8893f22b1228346659367e018d780b6ca5161
       #7 1.574 AlmaLinux 8 - BaseOS                            5.7 MB/s | 5.9 MB     00:01    
       #7 3.249 AlmaLinux 8 - AppStream                          14 MB/s |  10 MB     00:00    
       #7 5.197 AlmaLinux 8 - Extras                             62 kB/s |  18 kB     00:00    
       #7 5.849 Package curl-7.61.1-22.el8_6.4.x86_64 is already installed.
       #7 5.865 Dependencies resolved.
       #7 5.867 ================================================================================
       #7 5.867  Package              Arch        Version                     Repository   Size
       #7 5.867 ================================================================================
       #7 5.867 Installing:
       #7 5.867  openssh-clients      x86_64      8.0p1-13.el8                baseos      667 k
       #7 5.867  openssh-server       x86_64      8.0p1-13.el8                baseos      490 k
       #7 5.867  sudo                 x86_64      1.8.29-8.el8                baseos      925 k
       #7 5.867  which                x86_64      2.21-17.el8                 baseos       48 k
       #7 5.867 Installing dependencies:
       #7 5.867  libedit              x86_64      3.1-23.20170329cvs.el8      baseos      102 k
       #7 5.867  openssh              x86_64      8.0p1-13.el8                baseos      521 k
       #7 5.867 
       #7 5.867 Transaction Summary
       #7 5.867 ================================================================================
       #7 5.867 Install  6 Packages
       #7 5.867 
       #7 5.867 Total download size: 2.7 M
       #7 5.867 Installed size: 9.0 M
       #7 5.868 Downloading Packages:
       #7 6.220 (1/6): libedit-3.1-23.20170329cvs.el8.x86_64.rp 916 kB/s | 102 kB     00:00    
       #7 6.232 (2/6): openssh-8.0p1-13.el8.x86_64.rpm          4.1 MB/s | 521 kB     00:00    
       #7 6.279 (3/6): openssh-clients-8.0p1-13.el8.x86_64.rpm  3.8 MB/s | 667 kB     00:00    
       #7 6.283 (4/6): openssh-server-8.0p1-13.el8.x86_64.rpm   7.6 MB/s | 490 kB     00:00    
       #7 6.300 (5/6): sudo-1.8.29-8.el8.x86_64.rpm              13 MB/s | 925 kB     00:00    
       #7 6.304 (6/6): which-2.21-17.el8.x86_64.rpm             1.9 MB/s |  48 kB     00:00    
       #7 6.304 --------------------------------------------------------------------------------
       #7 6.304 Total                                           6.2 MB/s | 2.7 MB     00:00     
       #7 6.378 Running transaction check
       #7 6.403 Transaction check succeeded.
       #7 6.403 Running transaction test
       #7 6.448 Transaction test succeeded.
       #7 6.448 Running transaction
       #7 6.532   Preparing        :                                                        1/1 
       #7 6.549   Running scriptlet: openssh-8.0p1-13.el8.x86_64                            1/6 
       #7 6.615   Installing       : openssh-8.0p1-13.el8.x86_64                            1/6 
       #7 6.637   Installing       : libedit-3.1-23.20170329cvs.el8.x86_64                  2/6 
       #7 6.711   Installing       : openssh-clients-8.0p1-13.el8.x86_64                    3/6 
       #7 6.721   Running scriptlet: openssh-server-8.0p1-13.el8.x86_64                     4/6 
       #7 6.787   Installing       : openssh-server-8.0p1-13.el8.x86_64                     4/6 
       #7 6.797   Running scriptlet: openssh-server-8.0p1-13.el8.x86_64                     4/6 
       #7 6.813   Installing       : which-2.21-17.el8.x86_64                               5/6 
       #7 6.903   Installing       : sudo-1.8.29-8.el8.x86_64                               6/6 
       #7 6.915   Running scriptlet: sudo-1.8.29-8.el8.x86_64                               6/6 
       #7 6.997   Verifying        : libedit-3.1-23.20170329cvs.el8.x86_64                  1/6 
       #7 6.997   Verifying        : openssh-8.0p1-13.el8.x86_64                            2/6 
       #7 6.997   Verifying        : openssh-clients-8.0p1-13.el8.x86_64                    3/6 
       #7 6.997   Verifying        : openssh-server-8.0p1-13.el8.x86_64                     4/6 
       #7 6.997   Verifying        : sudo-1.8.29-8.el8.x86_64                               5/6 
       #7 6.998   Verifying        : which-2.21-17.el8.x86_64                               6/6 
       #7 7.016 
       #7 7.016 Installed:
       #7 7.016   libedit-3.1-23.20170329cvs.el8.x86_64    openssh-8.0p1-13.el8.x86_64          
       #7 7.016   openssh-clients-8.0p1-13.el8.x86_64      openssh-server-8.0p1-13.el8.x86_64   
       #7 7.016   sudo-1.8.29-8.el8.x86_64                 which-2.21-17.el8.x86_64             
       #7 7.016 
       #7 7.016 Complete!
       #7 DONE 7.1s
       
       #8 [ 4/16] RUN [ -f "/etc/ssh/ssh_host_rsa_key" ] || ssh-keygen -t rsa -f /etc/ssh/ssh_host_rsa_key -N ''
       #8 sha256:39771afba1683b68660f1dbedb231338d79b70832c607a1e7a31c5831baa00d6
       #8 0.449 Generating public/private rsa key pair.
       #8 0.449 Your identification has been saved in /etc/ssh/ssh_host_rsa_key.
       #8 0.449 Your public key has been saved in /etc/ssh/ssh_host_rsa_key.pub.
       #8 0.449 The key fingerprint is:
       #8 0.449 SHA256:L5rVQAWZcYHGCE6OH37WPnOA/FgDW6Creej6sjnvegc root@buildkitsandbox
       #8 0.449 The key's randomart image is:
       #8 0.449 +---[RSA 3072]----+
       #8 0.449 |    o..oo*+.     |
       #8 0.449 |   = ...*o       |
       #8 0.449 |  . = ..o        |
       #8 0.449 |   o + B         |
       #8 0.449 |    + * S        |
       #8 0.449 |  E+ o = *       |
       #8 0.449 |  +.. . B +      |
       #8 0.449 |oo...  + =       |
       #8 0.449 |=@=.  o          |
       #8 0.449 +----[SHA256]-----+
       #8 DONE 0.5s
       
       #9 [ 5/16] RUN [ -f "/etc/ssh/ssh_host_dsa_key" ] || ssh-keygen -t dsa -f /etc/ssh/ssh_host_dsa_key -N ''
       #9 sha256:9e3ae7ad70345ca53355692eb3549d23511bfab8b28c72a39a87542e6fc351c3
       #9 0.381 Generating public/private dsa key pair.
       #9 0.381 Your identification has been saved in /etc/ssh/ssh_host_dsa_key.
       #9 0.381 Your public key has been saved in /etc/ssh/ssh_host_dsa_key.pub.
       #9 0.381 The key fingerprint is:
       #9 0.381 SHA256:IVRIf4LMO4OMnaVjYMghdft86p/dNzsKOPgxDfyjjZQ root@buildkitsandbox
       #9 0.381 The key's randomart image is:
       #9 0.381 +---[DSA 1024]----+
       #9 0.381 |.o. ..oo.        |
       #9 0.381 |o... =.o         |
       #9 0.381 |..o . * + .      |
       #9 0.381 | . = B + +       |
       #9 0.381 |  . O * S        |
       #9 0.381 |   . . * *       |
       #9 0.381 |      o E =      |
       #9 0.381 |     . o X +  +  |
       #9 0.381 |      ..* o oo.+ |
       #9 0.381 +----[SHA256]-----+
       #9 DONE 0.4s
       
       #10 [ 6/16] RUN if ! getent passwd kitchen; then                   useradd -d /home/kitchen -m -s /bin/bash -p '*' kitchen;                 fi
       #10 sha256:c913dc5ce36f8861082b0da225e06c964cc9d3ab737daab96d20be7481388017
       #10 DONE 0.3s
       
       #11 [ 7/16] RUN echo "kitchen ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers.d/kitchen
       #11 sha256:11ba0f6398b5579460d016be10b50be804f1a8d63f352d79060c4f34af717740
       #11 DONE 0.3s
       
       #12 [ 8/16] RUN echo "Defaults !requiretty" >> /etc/sudoers.d/kitchen
       #12 sha256:d7a683356ba58bdd4f17d4165df8872435ea44328f8322334de6248d27da6241
       #12 DONE 0.3s
       
       #13 [ 9/16] RUN mkdir -p /home/kitchen/.ssh
       #13 sha256:3a8f112bec96fcd19ece51ca6ba9295568b92ee84f9218773dc5ecdbd8f7e83d
       #13 DONE 0.4s
       
       #14 [10/16] RUN chown -R kitchen /home/kitchen/.ssh
       #14 sha256:986fb43de4c23b006a85d078c8fc1ee595595eb92d55575ff2d9fee64de2be80
       #14 DONE 0.4s
       
       #15 [11/16] RUN chmod 0700 /home/kitchen/.ssh
       #15 sha256:d1c0797695bf03c862d51e54a8b1ba3c2afbf112bb40221eaa09248b66f09b3b
       #15 DONE 0.3s
       
       #16 [12/16] RUN touch /home/kitchen/.ssh/authorized_keys
       #16 sha256:f5555c80673476ff340b4108bc2354d71c0b2e6b1de089196c72c65f3c51d9d9
       #16 DONE 0.4s
       
       #17 [13/16] RUN chown kitchen /home/kitchen/.ssh/authorized_keys
       #17 sha256:56ed8c7089d4a6431644432800ac4c61ecb81fce3dee1f111c29a510c68d7197
       #17 DONE 0.4s
       
       #18 [14/16] RUN chmod 0600 /home/kitchen/.ssh/authorized_keys
       #18 sha256:de606f840a71b15cbe85f0a02ccfec59f10b8dd22200874c1c6e2cc643057ab7
       #18 DONE 0.3s
       
       #19 [15/16] RUN mkdir -p /run/sshd
       #19 sha256:eda2c4300d3ba503c419ec474bd1ee57b923504f9c489dba92ae4c5d76dcb23e
       #19 DONE 0.4s
       
       #20 [16/16] RUN echo ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCeNNln+Jjid1GYvWtUoCS6eg+VQn2ODjaH8YnDElUsFmVFTNd6Z8tWRyCnnuA16GXm2CKCsaxv+laisXldVfO0Fz0jW6DjW0kxzB6dejRfYh8RmGkTUbXE1BR5fxl/cx9lVMM9LMMco94QLjlZzmY+LzUodoI7IK5w4Dj832E0tDn8jXB6Jm0Fezahhq2xCHcyYotI8muof0S/MhiRf9xwrR+k8Ruhw/H3goYK2v5qjHJoCAx/kW+V9xU5nK/L+uTV4f4JyQ98uPvUwWbZQdFugyfdDzJCf3f6/11F3Wbay09unrEBamzgGJnAtoHyNytc26br0lMXwi/QMOIMbNxT kitchen_docker_key >> /home/kitchen/.ssh/authorized_keys
       #20 sha256:17f29f0b60b5a0293942d9663241b0884d1b3d5086146067294e7d6f8233b42d
       #20 DONE 0.4s
       
       #21 exporting to image
       #21 sha256:e8c613e07b0b7ff33893b694f7759a10d42e180f2b4dc349fb57dc6b71dcab00
       #21 exporting layers
       #21 exporting layers 0.5s done
       #21 writing image sha256:c890f4e1867d924d001f960acfa3ee6550b6cca2188e81df677e8bc4b90b354b done
       #21 DONE 0.5s
       3404b1be7ae8c8464a6565712f54c9f53a1e6d6896ba63e0eaa97927e84f268d
       0.0.0.0:50612
       Finished creating <default-almalinux-8> (0m14.85s).
-----> Converging <default-almalinux-8>...
       Preparing files for transfer
       Preparing dna.json
       Policyfile, Berksfile, cookbooks/, or metadata.rb not found so Chef Infra Client will run, but do nothing. Is this intended?
       Removing non-cookbook files before transfer
       Preparing validation.pem
       Preparing client.rb
       [Docker] Executing command on container
-----> Installing Chef install only if missing package
       Downloading https://omnitruck.chef.io/install.sh to file /tmp/install.sh
       Trying curl...
       Download complete.
       el 8 x86_64
       Getting information for chef stable  for el...
       downloading https://omnitruck.chef.io/stable/chef/metadata?v=&p=el&pv=8&m=x86_64
         to file /tmp/install.sh.29/metadata.txt
       trying curl...
       sha1	25ba625102b3d8d0200f923afe7ccc296945f20a
       sha256	c02dea79437b38a8b9e6f4c8d984318b16005ee8ade724708d4312e267d98dbb
       url	https://packages.chef.io/files/stable/chef/17.10.3/el/8/chef-17.10.3-1.el8.x86_64.rpm
       version	17.10.3
       downloaded metadata file looks valid...
       downloading https://packages.chef.io/files/stable/chef/17.10.3/el/8/chef-17.10.3-1.el8.x86_64.rpm
         to file /tmp/install.sh.29/chef-17.10.3-1.el8.x86_64.rpm
       trying curl...
       Comparing checksum with sha256sum...
       
       WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
       
       You are installing a package without a version pin.  If you are installing
       on production servers via an automated process this is DANGEROUS and you will
       be upgraded without warning on new releases, even to new major releases.
       Letting the version float is only appropriate in test, development or
       CI/CD environments.
       
       WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING
       
       Installing chef 
       installing with rpm...
       warning: /tmp/install.sh.29/chef-17.10.3-1.el8.x86_64.rpm: Header V4 DSA/SHA1 Signature, key ID 83ef826a: NOKEY
       Verifying...                          ########################################
       Preparing...                          ########################################
       Updating / installing...
       chef-17.10.3-1.el8                    ########################################
       Thank you for installing Chef Infra Client! For help getting started visit https://learn.chef.io
       [Docker] Executing command on container
       Transferring files to <default-almalinux-8>
       [Docker] Executing command on container
       +---------------------------------------------+
       ✔ 2 product licenses accepted.
       +---------------------------------------------+
       Chef Infra Client, version 17.10.3
       Patents: https://www.chef.io/patents
       Infra Phase starting
       [2022-09-13T17:48:41+00:00] WARN: Plugin Network: unable to detect ipaddress
       [2022-09-13T17:48:42+00:00] ERROR: shard_seed: Failed to get dmi property serial_number: is dmidecode installed?
       Creating a new client identity for default-almalinux-8 using the validator key.
       Resolving cookbooks for run list: []
       Synchronizing cookbooks:
       Installing cookbook gem dependencies:
       Compiling cookbooks...
       Loading Chef InSpec profile files:
       Loading Chef InSpec input files:
       Loading Chef InSpec waiver files:
       [2022-09-13T17:48:43+00:00] WARN: Node default-almalinux-8 has an empty run list.
       Converging 0 resources
       
       Running handlers:
       Running handlers complete
       Infra Phase complete, 0/0 resources updated in 01 seconds
       Downloading files from <default-almalinux-8>
       Finished converging <default-almalinux-8> (0m9.85s).
-----> Setting up <default-almalinux-8>...
       Finished setting up <default-almalinux-8> (0m0.00s).
-----> Verifying <default-almalinux-8>...
WARNING: Nokogiri was built against libxml version 2.9.4, but has dynamically loaded 2.9.13
       Loaded linux-patch-baseline 
[2022-09-13T12:48:48-05:00] WARN: Cannot find a UUID for your node.

Profile: DevSec Linux Patch Benchmark (linux-patch-baseline)
Version: 0.6.8
Target:  docker://3404b1be7ae8c8464a6565712f54c9f53a1e6d6896ba63e0eaa97927e84f268d

  ✔  verify-patches: Operating system is up-to-date
     ✔  Linux Update is expected to be uptodate
  ↺  patches: All operating system package updates are installed
     ↺  Skipped control due to only_if condition.


Profile Summary: 1 successful control, 0 control failures, 1 control skipped
Test Summary: 1 successful, 0 failures, 1 skipped
       Finished verifying <default-almalinux-8> (0m5.07s).
-----> Test Kitchen is finished. (0m32.08s)
```